src/NServiceBus.Core/Routing/UnicastPublishRouter.cs

```diff
@@ -26,7 +28,24 @@ public async Task<IEnumerable<UnicastRoutingStrategy>> Route(Type messageType, I
 
             var subscribers = await GetSubscribers(publishContext, typesToRoute).ConfigureAwait(false);
 
-            return SelectDestinationsForEachEndpoint(publishContext, distributionPolicy, subscribers);
+            var destinations = SelectDestinationsForEachEndpoint(publishContext, distributionPolicy, subscribers);
+
+            WarnIfNoSubscribersFound(messageType, destinations.Count);
+
+            return destinations;
+        }
+
+        void WarnIfNoSubscribersFound(Type messageType, int subscribersFound)
+        {
+            if (subscribersFound == 0)
+            {
+                eventsWithoutSubscribers.GetOrAdd(messageType.FullName, @event =>
```

Comment:
Nitpick:

Would using https://docs.microsoft.com/en-us/dotnet/api/system.collections.concurrent.concurrentdictionary-2.tryadd

```
if(eventsWithoutSubscribers.TryAdd(messageType.FullName, true))
{
    logger.WarnFormat("No subscribers found for the event of type {0}.", @event);
}
```

Be more intention revealing?

Reply:
Given we're going with a debug log, going to resolve this with no-op.